### PR TITLE
[core] Add more message when accessing a dead actor.

### LINKED ISSUE #30189: [Core] improve error message on calling a dead actor
### What happened + What you expected to happen

Today, when you call an actor method which belongs to a dead cluster yield following error message.
There is an opportunity to improve the error message.

```
  File "/home/ray/anaconda3/lib/python3.9/site-packages/ray/actor.py", line 138, in remote
    return self._remote(args, kwargs)
  File "/home/ray/anaconda3/lib/python3.9/site-packages/ray/util/tracing/tracing_helper.py", line 425, in _start_span
    return method(self, args, kwargs, *_args, **_kwargs)
  File "/home/ray/anaconda3/lib/python3.9/site-packages/ray/actor.py", line 184, in _remote
    return invocation(args, kwargs)
  File "/home/ray/anaconda3/lib/python3.9/site-packages/ray/actor.py", line 171, in invocation
    return actor._actor_method_call(
  File "/home/ray/anaconda3/lib/python3.9/site-packages/ray/actor.py", line 1170, in 
```

### Versions / Dependencies

latest

### Reproduction script

N/A

### Issue Severity

Low: It annoys or frustrates me.

### PR DESCRIPTION
**Pull request by fishbone**
## Why are these changes needed?
When any ray worker access the actor which doesn't belong to the cluster, it'll crash. This is really bad user experience because:

- The user will lose ray worker or driver.
- The error message is not very useful.

This PR fixed it by avoiding crash and throw exception instead. When the user accidentially access an actor which doesn't belong to this cluster, it'll get an Exception with message: `It might be dead or it's from a different cluster`.

This PR also deleted some code that's not useful.

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: BUILD.bazel [hunk 1]
```diff
@@ -479,29 +479,6 @@ cc_library(
     ],
 )
 
-# This header is used to warp some internal code so we can reduce suspicious
-# symbols export.
-cc_library(
-    name = "exported_internal",
-    srcs = glob(
-        [
-            "src/ray/internal/internal.cc",
```
Comment (rkooo567) [inline BUILD.bazel:488]: Guess it is unrelated changes? Looks okay to include in this PR to me.

File: src/ray/core_worker/core_worker.cc [hunk 2]
```diff
@@ -2175,17 +2175,24 @@ Status CoreWorker::WaitPlacementGroupReady(const PlacementGroupID &placement_gro
...
     const RayFunction &function,
     const std::vector<std::unique_ptr<TaskArg>> &args,
-    const TaskOptions &task_options) {
+    const TaskOptions &task_options,
+    std::optional<std::vector<rpc::ObjectReference>> &task_returns) {
   absl::ReleasableMutexLock lock(&actor_task_mutex_);
+  task_returns = std::nullopt;
+  if (!direct_actor_submitter_->CheckActorExists(actor_id)) {
+    return Status::NotFound(
+        "Can't find this actor. It might be dead or it's from a different cluster");
```
Comment (rkooo567) [inline src/ray/core_worker/core_worker.cc:2188]: Can you add an actor id to the error message?

File: src/ray/core_worker/core_worker.cc [hunk 3]
```diff
@@ -2175,17 +2175,24 @@ Status CoreWorker::WaitPlacementGroupReady(const PlacementGroupID &placement_gro
...
+  }
   /// Check whether backpressure may happen at the very beginning of submitting a task.
   if (direct_actor_submitter_->PendingTasksFull(actor_id)) {
     RAY_LOG(DEBUG) << "Back pressure occurred while submitting the task to " << actor_id
                    << ". " << direct_actor_submitter_->DebugString(actor_id);
-    return std::nullopt;
+    return Status::OutOfResource(
+        "Too many tasks pending to be executed. Please try later");
```
Comment (rkooo567) [inline src/ray/core_worker/core_worker.cc:2195]: Can you print the current number of pending tasks?

**Review by rkooo567 — APPROVED** [review]
LGTM. just NITs for error message improvement

File: BUILD.bazel (hunk 1 above)
Comment (iycheng) [reply BUILD.bazel:488]: Actually it is... There is some method used in this file got impacted by this PR. So when I compile it just failed :( 
I notice it seems not useful, so just delete that file.
